module/platform/NUSense/HardwareIO/src/HardwareIO.cpp

```diff
@@ -0,0 +1,222 @@
+                servo.present_current       = val.present_current;
+                servo.present_velocity      = val.present_velocity;
+                servo.present_position      = val.present_position;
+                servo.goal_PWM              = val.goal_pwm;
+                servo.goal_current          = val.goal_current;
+                servo.goal_velocity         = val.goal_velocity;
+                servo.goal_position         = val.goal_position;
+                servo.voltage               = val.voltage;
+                servo.temperature           = val.temperature;
+                servo.profile_acceleration  = 0;  // not present in NUSense message
+                servo.profile_velocity      = 0;  // not present in NUSense message
+
+                // Add the offsets and switch the direction.
+                servo.present_position *= nugus.servo_direction[val.id - 1];
+                servo.present_position += nugus.servo_offset[val.id - 1];
+            }
+
+            if (log_level <= NUClear::DEBUG) {
+                log<NUClear::DEBUG>(
+                    fmt::format("\nIMU Data\n"
+                                "\tAccel(xyz): {} - {} - {}\n"
+                                "\t Gyro(xyz): {} - {} - {}\n ",
+                                data.imu.accel.x,
+                                data.imu.accel.y,
+                                data.imu.accel.z,
+                                data.imu.gyro.x,
+                                data.imu.gyro.y,
+                                data.imu.gyro.z));
+
+                log<NUClear::DEBUG>("Logging servo states...");
+
+                for (const auto& [key, val] : data.servo_map) {
+                    log<NUClear::DEBUG>(fmt::format("      key: {}", key));
+
+                    log<NUClear::DEBUG>(fmt::format("       id: {}", val.id));
+                    log<NUClear::DEBUG>(fmt::format("   hw_err: {}", val.hardware_error));
+                    log<NUClear::DEBUG>(fmt::format("torque_en: {}", val.torque_enabled));
+                    log<NUClear::DEBUG>(fmt::format("     ppwm: {}", val.present_pwm));
+                    log<NUClear::DEBUG>(fmt::format("    pcurr: {}", val.present_current));
+                    log<NUClear::DEBUG>(fmt::format("    pvelo: {}", val.present_velocity));
+                    log<NUClear::DEBUG>(fmt::format("     ppos: {}", val.present_position));
+                    log<NUClear::DEBUG>(fmt::format("     gpwm: {}", val.goal_pwm));
+                    log<NUClear::DEBUG>(fmt::format("    gcurr: {}", val.goal_current));
+                    log<NUClear::DEBUG>(fmt::format("    gvelo: {}", val.goal_velocity));
+                    log<NUClear::DEBUG>(fmt::format("     gpos: {}", val.goal_position));
+                    log<NUClear::DEBUG>(fmt::format("  voltage: {}", val.voltage));
+                    log<NUClear::DEBUG>(fmt::format("     temp: {}", val.temperature));
+                }
+            }
```

Comment:
theres no need to wrap this in an if

Reply:
As a counter point to my other comment with the trace line, this one doesn't need the if statement